ibm/service/vpc/resource_ibm_is_security_group_rule.go

```diff
@@ -613,29 +613,23 @@ func parseIBMISSecurityGroupRuleDictionary(d *schema.ResourceData, tag string, s
 	if icmpInterface, ok := d.GetOk("icmp"); ok {
 		if icmpInterface.([]interface{})[0] != nil {
```

Comment:
since type and code both optional if user provide empty icmp{} block ill accessing 0th element will have nil pinter exeception can you check length and access

Reply:
if icmpInterface, ok := d.GetOk("icmp"); ok {     This if will not pass if we pass  icmp{}..
Tested with the below block and its working -

resource "ibm_is_security_group_rule" "testacc_security_group_rule_icmp_any" {
  group     = ibm_is_security_group.testacc_security_group.id
  direction = "inbound"
  remote    = "127.0.0.1"
  icmp {
  }
}

<img width="764" alt="Screenshot 2023-04-20 at 11 38 12 AM" src="https://user-images.githubusercontent.com/127872893/233273187-cfaf64cf-6ed3-4b19-8bcf-a234145b712b.png">